lib/sles4sap/console_redirection.pm

```diff
@@ -175,6 +175,11 @@ from serial console. VM ID is collected by opening 'log-console' which is not re
 sub check_serial_redirection {
     # Do not select serial console if it is already done. This avoids log pollution and speeds up process.
     if (is_serial_terminal()) {
+        # Teminate the process when scirpts timed out
+        unless (wait_serial(serial_term_prompt())) {
```

Comment:
This part is covered by Unit testing in os-autoinst-distri-opensuse/t/19_console_redirection.t

Reply:
What is this part supposed to do ?

Reply:
> What is this part supposed to do ?

Waits 30 seconds for the serial prompt. If it's not seen, then `wait_serial` times out and returns `undef` which would take the test code into the block in L180-181. That sends first a Ctrl-C and then a `\n` to force the prompt to appear.